Rubberduck.Core/AutoComplete/Service/SelfClosingPairCompletionService.cs

```diff
@@ -44,40 +41,63 @@ public CodeString Execute(SelfClosingPair pair, CodeString original, char input,
                 return HandleClosingChar(pair, original);
             }
 
-            return default;
+            if (input == '\b')
+            {
+                return Execute(pair, original, Keys.Back);
+            }
+
+            return null;
         }
 
         public CodeString Execute(SelfClosingPair pair, CodeString original, Keys input)
         {
+            if (original.IsComment)
+            {
+                return null;
+            }
+
             if (input == Keys.Back)
             {
                 return HandleBackspace(pair, original);
             }
 
-            return default;
+            return null;
         }
 
         private CodeString HandleOpeningChar(SelfClosingPair pair, CodeString original)
         {
             var nextPosition = original.CaretPosition.ShiftRight();
             var autoCode = new string(new[] { pair.OpeningChar, pair.ClosingChar });
-            var lines = original.Code.Split('\n');
+            var lines = original.Lines;
             var line = lines[original.CaretPosition.StartLine];
-            lines[original.CaretPosition.StartLine] = line.Insert(original.CaretPosition.StartColumn, autoCode);
+            lines[original.CaretPosition.StartLine] = string.IsNullOrEmpty(original.Code) 
+                    ? autoCode 
+                    : original.CaretPosition.StartColumn == line.Length 
+                        ? line + autoCode 
+                        : line.Insert(original.CaretPosition.StartColumn, autoCode);
```

Comment:
This nested ternary is the core of what we're doing here. I think we can give it some space :)

Consider:

```
string resultCode;
if (string.isNullOrEmpty(original.Code)
{
    resultCode = autoCode;
} 
else if (original.CaretPosition.StartColumn == line.Length)
{
    resultCode = line + autoCode;
}
else 
{
    resultCode = line.Insert(original.CaretPosition.StartColumn, autoCode);
}
lines[original.CaretPosition.StartLine] = resultCode;
```

This is much less compressed and separates logic from data